source/common/upstream/cluster_manager_impl.h

```diff
@@ -285,6 +286,42 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
   ClusterUpdateCallbacksHandlePtr
   addThreadLocalClusterUpdateCallbacks(ClusterUpdateCallbacks&) override;
 
+  class OdCdsApiHandleImpl : public std::enable_shared_from_this<OdCdsApiHandleImpl>,
```

Comment:
Can this be at least protected, so unit test could use it?

Reply:
You can make the test a friend if you need to test private details, though optimally the tests would just go through the main CM.

Reply:
I've wanted to make it protected, so the test can construct a own handle that contains a mock odcds, instead of the real one. So when handle->requestOnDemandClusterDiscovery, a real CM code will be tested, but it will call mock's `odcds.updateOnDemand`. I'll push my updates soon, so you will see what I mean.